core/src/tile/tileManager.cpp

```diff
@@ -3,237 +3,350 @@
+        {
+            auto setTilesIter = m_tileSet.begin();
+            auto visTilesIter = visibleTiles.begin();
+
+            while (visTilesIter != visibleTiles.end()) {
+
+                auto& visTileId = *visTilesIter;
+                auto& curTileId = setTilesIter == m_tileSet.end() ? NOT_A_TILE : setTilesIter->first;
+
+                if (visTileId == curTileId) {
+                    if (setTilesIter->second->state() == MapTile::None) {
+                        logMsg("[%d, %d, %d] - Enqueue\n", curTileId.z, curTileId.x, curTileId.y);
+                        m_loadQueue.push_back(curTileId);
+                    }
+
+                    // tiles in both sets match
+                    ++setTilesIter;
+                    ++visTilesIter;
+
+                } else if (curTileId == NOT_A_TILE || visTileId < curTileId) {
+                    // tileSet is missing an element present in visibleTiles
+                    addTile(visTileId);
+                    ++visTilesIter;
+
                 } else {
+                    // visibleTiles is missing an element present in tileSet (handled below)
                     ++setTilesIter;
                 }
-            } else if (*visTilesIter < setTilesIter->first) {
-                // tileSet is missing an element present in visibleTiles (shouldn't occur)
-                ++visTilesIter;
-            } else {
-                // tiles in both sets match, move on
-                ++setTilesIter;
-                ++visTilesIter;
+            }
+        }
+
+        // Loop over tileSet and remove any tiles that are neither visible nor proxies
+        {
+            auto setTilesIter = m_tileSet.begin();
+            auto visTilesIter = visibleTiles.begin();
+
+            while (setTilesIter != m_tileSet.end()) {
+
+                auto& visTileId = visTilesIter == visibleTiles.end() ? NOT_A_TILE : *visTilesIter;
+                auto& curTileId = setTilesIter->first;
+
+                if (visTileId == curTileId) {
+                    // tiles in both sets match
+                    ++setTilesIter;
+                    ++visTilesIter;
+
+                } else if (visTileId == NOT_A_TILE || curTileId < visTileId) {
+                    // remove element from tileSet not present in visibleTiles
+                    const auto& tile = setTilesIter->second;
+                    if (tile->getProxyCounter() <= 0) {
+                        removeTile(setTilesIter);
+                    } else {
+                        ++setTilesIter;
+                    }
+                } else {
+                    // tileSet is missing an element present in visibleTiles (shouldn't occur)
+                    ++visTilesIter;
+                }
+            }
+        }
+    }
+
+    if (!m_loadQueue.empty() && m_loadPending < MAX_PARALLEL_DOWNLOADS) {
+        glm::dvec2 center(m_view->getPosition().x, -m_view->getPosition().y);
+
+        m_loadQueue.sort([&](const TileID& a, const TileID& b) {
```

Comment:
Numbers are good for sorting.

We should be careful with polling implementation, if a bunch of tiles (tilted view) are being fetched and with std::async we can see bad thread performance because of trashing and context switches. Also platforms like raspberry pi might not perform well with this.